notebooks/nessie-iceberg-demo-nba.ipynb

```diff
@@ -0,0 +1,629 @@
+    "This demo showcases how to use Nessie Python API along with Spark3 from Iceberg\n",
+    "\n",
+    "Initialize Pyspark\n",
+    "----------------------------------------------\n",
+    "To get started, we will first have to do a few setup steps that give us everything we need\n",
+    "to get started with Nessie. In case you're interested in the detailed setup steps for Spark, you can check out the [docs](https://projectnessie.org/tools/spark/).\n",
+    "\n",
+    "The Binder server has downloaded spark and some data for us as well as started a Nessie server in the background. All we have to do is start Spark.\n",
+    "\n",
+    "The below cell starts a local Spark session with parameters needed to configure Nessie. Each config option is followed by a comment explaining its purpose.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import findspark\n",
+    "from pyspark.sql import *\n",
+    "from pyspark import SparkConf\n",
+    "findspark.init()\n",
+    "\n",
+    "conf = SparkConf()\n",
+    "# we need iceberg libraries and the nessie sql extensions\n",
+    "conf.set(\"spark.jars.packages\", \"org.apache.iceberg:iceberg-spark3-runtime:0.12.0,org.projectnessie:nessie-spark-extensions:0.9.0\")\n",
+    "# ensure python <-> java interactions are w/ pyarrow\n",
+    "conf.set(\"spark.sql.execution.pyarrow.enabled\", \"true\")\n",
+    "# create catalog dev_catalog as an iceberg catalog\n",
+    "conf.set(\"spark.sql.catalog.dev_catalog\", \"org.apache.iceberg.spark.SparkCatalog\")\n",
+    "# tell the dev_catalog that its a Nessie catalog\n",
+    "conf.set(\"spark.sql.catalog.dev_catalog.catalog-impl\", \"org.apache.iceberg.nessie.NessieCatalog\")\n",
+    "# set the location for Nessie catalog to store data. Spark writes to this directory\n",
+    "conf.set(\"spark.sql.catalog.dev_catalog.warehouse\", 'file://' + os.getcwd() + '/spark_warehouse')\n",
+    "# set the location of the nessie server. In this demo its running locally. There are many ways to run it (see https://projectnessie.org/try/)\n",
+    "conf.set(\"spark.sql.catalog.dev_catalog.uri\", \"http://localhost:19120/api/v1\")\n",
+    "# default branch for Nessie catalog to work on\n",
+    "conf.set(\"spark.sql.catalog.dev_catalog.ref\", \"main\")\n",
+    "# use no authorization. Options are NONE AWS BASIC and aws implies running Nessie on a lambda\n",
+    "conf.set(\"spark.sql.catalog.dev_catalog.auth_type\", \"NONE\")\n",
+    "# enable the extensions for both Nessie and Iceberg\n",
+    "conf.set(\"spark.sql.extensions\", \"org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions,org.projectnessie.spark.extensions.NessieSparkSessionExtensions\")\n",
+    "# finally, start up the Spark server\n",
+    "spark = SparkSession.builder.config(conf=conf).getOrCreate()\n",
+    "print(\"Spark Running\")"
```

Comment:
Can you add a `spark.sql("LIST REFERENCES IN dev_catalog").toPandas() ` so the first "real" interaction with Spark/Nessie isn't slow?